### PR TITLE
+ Support subtrees

### PR DESCRIPTION
**Pull request by tetreum**
Allows handling large datasets without suffering (https://github.com/morris/microdb/issues/9)

``` php
$this->options["subtree"]
```

Would be an integer.

### CODE REVIEW

File: src/MicroDB/Database.php [hunk 1]
```diff
@@ -19,6 +19,11 @@ class Database
      * @var int
      */
     protected $mode;
+    
+    /**
+     * @var array
+     */
+    protected $options = array();
```
Comment (rickerd) [inline src/MicroDB/Database.php]: Can you use the new 5.4 array syntax?
And also update composer.json to meet the new php 5.6 standard?

File: src/MicroDB/Database.php [hunk 2]
```diff
@@ -36,12 +41,13 @@ class Database
      * @param string $path
      * @param int $mode
      */
-    public function __construct($path, $mode = 0775)
+    public function __construct($path, $mode = 0775, $options = array())
```
Comment (rickerd) [inline src/MicroDB/Database.php]: New syntax here too please

**Comment by tetreum** [comment]
I didn't apply the new syntax because the other arrays were set with the older one. Give me a sec.

**Comment by tetreum** [comment]
¿Couldn't we take advantage of this commit to implement travis autotesting?

**Comment by rickerd** [comment]
Ofcourse, we can do all sorts of things.

**Comment by tetreum** [comment]
We can use this yml with a small change on script part:
https://github.com/tetreum/kat/blob/master/.travis.yml

**Comment by rickerd** [comment]
I need some help from @morris to enable travis.
The .travis.yml file is ready (i think)

http://pastebin.com/D45dXzJt

**Comment by tetreum** [comment]
¡Cool!

**Pull request merged**